Common/Algorithm/Framework/Alphas/Insight.cs

```diff
@@ -217,11 +218,17 @@ public static Guid Group(params Insight[] insights)
                 }
 
                 insight.GroupId = groupId;
-            }
 
-            return groupId;
+                yield return insight;
+            }
         }
 
+        /// <summary>
+        /// Creates a new, unique group id and sets it on each insight
+        /// </summary>
+        /// <param name="insight">The insight to be grouped</param>
+        public static IEnumerable<Insight> Group(Insight insight) => Group(new[] { insight });
```

Comment:
Is this necessary? The `params` modifier should do this by default? Is this added for python convenience?

Reply:
Yes. This is due to pythonnet inability to solve this:
```python
Insight.Group(insight)
```
as it says that it cannot find an overload. 
I could take a look at pythonnet and solve this issue.

Reply:
But is that required? a Group method for one insight? I think all other places we use C# `params` we use python `[]` array notation. I'm not against adding this, just wondering if it adds much value, that's all.